digitalocean/resource_digitalocean_kubernetes_node_pool.go

```diff
@@ -148,7 +198,17 @@ func resourceDigitalOceanKubernetesNodePoolRead(d *schema.ResourceData, meta int
 	d.Set("name", pool.Name)
 	d.Set("size", pool.Size)
 	d.Set("node_count", pool.Count)
+	d.Set("actual_node_count", pool.Count)
 	d.Set("tags", flattenTags(filterTags(pool.Tags)))
+	d.Set("auto_scale", pool.AutoScale)
+	d.Set("min_nodes", pool.MinNodes)
+	d.Set("max_nodes", pool.MaxNodes)
+
+	// Assign a node_count only if it's been set explicitly, since it's
+	// optional and we don't want to update with a 0 if it's not set.
+	if _, ok := d.GetOk("node_count"); ok {
+		d.Set("node_count", pool.Count)
+	}
 
 	if pool.Nodes != nil {
 		d.Set("nodes", flattenNodes(pool.Nodes))
```

Comment:
Doing a final round of testing before merging. I'm running into another issue. If you create a node pool with auto scaling and no nodes to begin with like so:

```
resource "digitalocean_kubernetes_cluster" "foobar" {
  name    = "test-cluster-01"
  region  = "lon1"
  version = "1.15.5-do.0"

  node_pool {
    name = "default"
    size  = "s-1vcpu-2gb"
    auto_scale = true
    node_count = 1
    min_nodes = 1
    max_nodes = 4
  }
}

resource "digitalocean_kubernetes_node_pool" "foobar" {
  name       = "test-pool"
  size       = "s-1vcpu-2gb"
  auto_scale = true
  min_nodes = 0
  max_nodes = 5
  cluster_id = "${digitalocean_kubernetes_cluster.foobar.id}"
}
```

The state ends up with `"nodes": null,` rather than an empty set (`"nodes": [],`). This leads to a diff on every plan/apply. Doing something along these lines fixes the issue:

```
	if pool.Nodes != nil {
		d.Set("nodes", flattenNodes(pool.Nodes))
	} else {
		d.Set("nodes", []godo.KubernetesNode{})
	}
```

Reply:
@snormore I push a commit to handle this case including a test: https://github.com/terraform-providers/terraform-provider-digitalocean/pull/307/commits/304ba94ed97797542f99753e90b67dade6dd3c96

Without the change, the test fails with `testing.go:569: Step 0 error: After applying this step, the plan was not empty`

With the change:

```
$ make testacc TESTARGS="-run=TestAccDigitalOceanKubernetesNodePool_WithEmptyNodePool"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccDigitalOceanKubernetesNodePool_WithEmptyNodePool -timeout 120m
?   	github.com/terraform-providers/terraform-provider-digitalocean	[no test files]
=== RUN   TestAccDigitalOceanKubernetesNodePool_WithEmptyNodePool
=== PAUSE TestAccDigitalOceanKubernetesNodePool_WithEmptyNodePool
=== CONT  TestAccDigitalOceanKubernetesNodePool_WithEmptyNodePool
--- PASS: TestAccDigitalOceanKubernetesNodePool_WithEmptyNodePool (273.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-digitalocean/digitalocean	273.179s
```

If you're good with that commit, I think we're ready to merged! :tada: Thanks for baring with me through multiple rounds of review.

Reply:
Awesome, thanks! 🙏 
Let's ⛵️